--- vlogger~	2006-11-24 02:45:46.366608000 +0100
+++ vlogger	2006-11-24 03:30:21.466608000 +0100
@@ -114,6 +114,13 @@
 -v
 	Prints version information.
 
+-x PREFIX
+	Accept potentially unsafe vhost names (useful if you replace
+%v with %{DOCUMENT_ROOT}e)
+
+-p PATH
+	Default logfile to use for unmatched vhosts
+
 =head1 DBI USAGE TRACKER
 
 	Vlogger can automatically keep track of per-virtualhost usage statistics in a
@@ -150,7 +157,7 @@
 
 # get command line options
 our %OPTS;
-getopts( 'f:t:s:hu:g:aeivr:d:', \%OPTS );
+getopts( 'f:t:s:hu:g:aeivp:x:r:d:', \%OPTS );
 
 # print out version
 if ( $OPTS{'v'} ) {
@@ -167,6 +174,10 @@
     exit;
 }
 
+unless ( $OPTS{'p'} ) {
+    $OPTS{'p'} = "default";
+}
+
 # log directory
 my $LOGDIR;
 if ( $ARGV[0] ) {
@@ -376,13 +387,15 @@
         my @this_line = split ( /\s/, $log_line );
         my ($vhost) = $this_line[0];
         my $reqsize = $this_line[10];
-        $vhost = lc($vhost) || "default";
+        $vhost = lc($vhost) || $OPTS{'p'};
         if (( $vhost =~ m#[/\\]# ) || ( $vhost !~ m#[a-zA-Z0-9\-]\.[a-zA-Z0-9\-\.]# )) {
-	    $vhost = "default"
-	}
+            unless ( $OPTS{'x'} && ( substr($vhost, 0, length($OPTS{'x'})) eq $OPTS{'x'} )) {
+                $vhost = $OPTS{'p'};
+            }
+        }
         $vhost =~ /(.*)/o;
         $vhost = $1;
-	$vhost = 'default' unless $vhost;
+	$vhost = $OPTS{'p'} unless $vhost;
 
         if ( $OPTS{'i'} ) {
             $reqsize = $this_line[1] + $this_line[2];
@@ -549,6 +562,8 @@
     print "  -r SIZE                     rotate when file reaches SIZE\n";
     print "  -d CONFIG                   use DBI usage tracker (see perldoc vlogger)\n";
     print "  -i                          extract mod_logio instead of filesize\n";
+    print "  -x PREFIX                   accept potentially unsafe vhost names\n";
+    print "  -p PATH                     default logging path for unknown vhosts\n";
     print "  -h                          display this help\n";
     print "  -v                          output version information\n\n";
     print "TEMPLATE may be a filename with Date::Format codes.  The default template\n";
